pennylane/devices/default_qutrit.py

```diff
@@ -138,6 +147,35 @@ def apply(self, operations, rotations=None, **kwargs):  # pylint: disable=argume
         for operation in rotations:
             self._state = self._apply_operation(self._state, operation)
 
+    def _apply_basis_state(self, state, wires):
+        """Initialize the state vector in a specified computational basis state.
+
+        Args:
+            state (array[int]): computational basis state of shape ``(wires,)``
+                consisting of 0s, 1s and 2s.
+            wires (Wires): wires that the provided computational state should be initialized on
+
+        Note: This function does not support broadcasted inputs yet.
+        """
+        # translate to wire labels used by device
+        device_wires = self.map_wires(wires)
+
+        # length of basis state parameter
+        n_basis_state = len(state)
+
+        if not set(state.tolist()).issubset({0, 1, 2}):
+            raise ValueError("QutritBasisState parameter must consist of 0, 1 or 2 integers.")
```

Comment:
I think this check would be more appropriate in the `__init__` method of `QutritBasisState`. The error shouldn't rely on the device, as it is due to an ill-defined operator as opposed to an ill-defined device.

As an example, if someone created a `QutritBasisState` instance outside a `QNode` or quantum tape with the parameter `[0, 1, 2, 3]`, an error wouldn't get thrown, but `QutritBasisState.state_vector` would likely not work.

Reply:
You are right, an error is not thrown until the `state_vector` method is called. Are you suggesting creating an ` __init__ ` that supers the one inherited from the `Operation` class? Then I should also change everything accordingly for the `Qubit BasisState` preparation and `default.qubit` as well right?